agent.py

```diff
@@ -136,8 +137,9 @@ def reload_configs(self, checks_to_reload=set()):
                 jmx_sd_configs = generate_jmx_configs(self._agentConfig, hostname)
         else:
             new_checksd = copy(self._checksd)
-
-            jmx_checks = [check for check in checks_to_reload if check in JMX_CHECKS]
+            auto_conf_is_jmx_checks = get_jmx_checks()
+            all_jmx_checks = JMX_CHECKS + auto_conf_is_jmx_checks
```

Comment:
I noticed that practically everywhere where we're calling `get_jmx_checks()` we then do:
```
all_jmx_checks = get_jmx_checks()
all_jmx_checks += JMX_CHECKS 
```

Maybe we should just do that in the actual function instead when called with the auto_conf paramenter set to `True`? Let me know if you think this doesn't make sense.

Also, I'm not sure if it might be a problem having duplicates (possibly not, but it would still be cleaner not to have them), so we could maybe use a set instead?